lib/logutils/firstandinterval.go

```diff
@@ -0,0 +1,296 @@
+			entry.Warning(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Error(args ...interface{}) {
+	if logger.level() >= logrus.ErrorLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Error(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Debugf(format string, args ...interface{}) {
+	if logger.level() >= logrus.DebugLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Debugf(format, args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Infof(format string, args ...interface{}) {
+	if logger.level() >= logrus.InfoLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Infof(format, args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Printf(format string, args ...interface{}) {
+	if entry := logger.logEntry(); entry != nil {
+		entry.Printf(format, args...)
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Warnf(format string, args ...interface{}) {
+	if logger.level() >= logrus.WarnLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Warnf(format, args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Warningf(format string, args ...interface{}) {
+	if logger.level() >= logrus.WarnLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Warningf(format, args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Errorf(format string, args ...interface{}) {
+	if logger.level() >= logrus.ErrorLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Errorf(format, args...)
+		}
+	}
+}
+
+// Entry Println family functions
+
+func (logger *FirstAndIntervalLogger) Debugln(args ...interface{}) {
+	if logger.level() >= logrus.DebugLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Debugln(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Infoln(args ...interface{}) {
+	if logger.level() >= logrus.InfoLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Infoln(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Println(args ...interface{}) {
+	if entry := logger.logEntry(); entry != nil {
+		entry.Println(args...)
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Warnln(args ...interface{}) {
+	if logger.level() >= logrus.WarnLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Warnln(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Warningln(args ...interface{}) {
+	if logger.level() >= logrus.WarnLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Warningln(args...)
+		}
+	}
+}
+
+func (logger *FirstAndIntervalLogger) Errorln(args ...interface{}) {
+	if logger.level() >= logrus.ErrorLevel {
+		if entry := logger.logEntry(); entry != nil {
+			entry.Errorln(args...)
+		}
+	}
+}
+
+// level returns the log level associated with the logger.  (copied from logrus since this is not a public method)
+func (logger *FirstAndIntervalLogger) level() logrus.Level {
+	return logrus.Level(atomic.LoadUint32((*uint32)(&logger.entry.Logger.Level)))
```

Comment:
I normally check against `logrus.GetLevel()` is this doing something different?

Reply:
Because I allowed the Logger to be passed in on the constructor I think we could therefore not use the logrus.GetLevel() since that always pulled from the standard logger.

Reply:
That said, I only added that to the constructor for testing ... so other suggestions welcome.